### PR TITLE
Sms template fixes

### PR DESCRIPTION
**Pull request by omarZrayka23**
allow delete , allow add newline and null

### CODE REVIEW

File: src/pages/sms-function-templates/index.js [hunk 1]
```diff
@@ -30,114 +26,106 @@ const SmsFunctionTemplate = () => {
...
+            name: templateName
+          },
+          ...rest
+        }))
+      });
     } catch (error) {
-      setErrorMessage(error.res)
 
-      return Promise.reject(error) // You can choose to reject the promise if an error occurs
+return Promise.reject(error);
```
Comment (hadichahine) [inline src/pages/sms-function-templates/index.js:98]: No need to do that. Remove catch block.

File: src/pages/sms-function-templates/index.js [hunk 2]
```diff
@@ -150,37 +138,27 @@ const SmsFunctionTemplate = () => {
...
-  })
 
-  const handleSubmit = () => {
-    smsFunctionTemplatesValidation.handleSubmit()
-  }
 
+  ]
+
+  const postSmsFunctionTemplates = (values) => {
+    console.log(initialValues)
```
Comment (hadichahine) [inline src/pages/sms-function-templates/index.js:157]: Remove.

File: src/pages/sms-function-templates/index.js [hunk 3]
```diff
@@ -193,48 +171,34 @@ const SmsFunctionTemplate = () => {
...
+              <Grid sx={{ width: '100%'  }}>
                 <Box sx={{ width: '100%'  }}>
                   <DataGrid
-                   height={height-150}
-                   onChange={value => smsFunctionTemplatesValidation.setFieldValue('rows', value)}
-                   value={smsFunctionTemplatesValidation.values.rows}
-                   error={smsFunctionTemplatesValidation.errors.rows}
+                   height={`calc(100vh - 150px)`}
+                   onChange={value => { console.log(value); formik.setFieldValue('rows', value)}}
+                   onCellEditStop={value => console.log(value, 'sms')}
```
Comment (hadichahine) [inline src/pages/sms-function-templates/index.js:189]: What's this? Datagrid shouldn't take this as a prop

File: src/pages/sms-function-templates/index.js [hunk 4]
```diff
@@ -193,48 +171,34 @@ const SmsFunctionTemplate = () => {
...
-              <Grid>
+              <Grid sx={{ width: '100%'  }}>
                 <Box sx={{ width: '100%'  }}>
                   <DataGrid
-                   height={height-150}
-                   onChange={value => smsFunctionTemplatesValidation.setFieldValue('rows', value)}
-                   value={smsFunctionTemplatesValidation.values.rows}
-                   error={smsFunctionTemplatesValidation.errors.rows}
+                   height={`calc(100vh - 150px)`}
+                   onChange={value => { console.log(value); formik.setFieldValue('rows', value)}}
```
Comment (hadichahine) [inline src/pages/sms-function-templates/index.js:188]: Remove log, and rename to something other than rows.

File: src/pages/clients-corporate-list/index.js [hunk 5]
```diff
@@ -24,36 +24,36 @@ import { useResourceQuery } from 'src/hooks/resource'
...
+  const { getRequest } = useContext(RequestsContext)
 
   //control
 
   const [editMode, setEditMode] = useState(null)
-  const [referenceRequired, setReferenceRequired] = useState(true)
-
 
   //stores
   const [errorMessage, setErrorMessage] = useState(null)
```
Comment (hadichahine) [inline src/pages/clients-corporate-list/index.js:34]: Do we need this?

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/clients-corporate-list/index.js (hunk 5 above)
Comment (omarZrayka23) [reply src/pages/clients-corporate-list/index.js:32]: no

File: src/components/Shared/DataGrid/components/resourcecombobox/edit.js [hunk 6]
```diff
@@ -18,20 +18,14 @@ return (
       dataGrid={true}
       readOnly={props?.readOnly}
       onChange={(e, value) => {
-        if(value)
+        const newValue = value || ''
         api.current.setEditCellValue({
           id,
           field,
-          value
+          newValue
```
Comment (hadichahine) [inline src/components/Shared/DataGrid/components/resourcecombobox/edit.js:25]: This won't work. Have you tested? Should pass value: value || ''

**Review by hadichahine — CHANGES_REQUESTED** [review]

File: src/pages/clients-corporate-list/index.js (hunk 5 above)
Comment (hadichahine) [reply src/pages/clients-corporate-list/index.js:32]: Remove pls

Comment (omarZrayka23) [reply src/pages/clients-corporate-list/index.js:32]: removed

**Pull request merged**